lib/redlock/client.rb

```diff
@@ -228,6 +258,22 @@ def lock_instances(resource, ttl, options)
       end
     end
 
+    def try_get_remaining_ttl(resource, value)
+      ttls, time_elapsed = timed do
+        @servers.map { |s| s.get_remaining_ttl resource, value }.compact
+      end
+
+      if ttls.size >= @quorum
+        min_ttl = ttls.sort.last(@quorum).min
```

Comment:
ok, I thought this would have a heavier footprint in the code than `.last(@quorum)` 😄 

Though, I'm now relatively confident that a "reduce" solution and `break` in the middle is in fact better. If we consider the "average" case to be when each server shows almost identical & constant roundtrip time, the `ttls` list will be ordered descending (`[50, 40, 30, 20, 10]`), so we'll usually ditch the values acquired last (and subtract a greater `time_elapsed` from the earlier values, so we don't gain anything).

Sorry for making you take this detour 😊